esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -3095,6 +3118,8 @@ void EsdlBindingImpl::transformGatewaysConfig( IPropertyTree* srvcfg, IPropertyT
         cfgIter->first();
         if( cfgIter->isValid() )
         {
+            const char* treeName = (altElementName && *altElementName ? altElementName : "Gateway");
```

Comment:
I think if altElementName is null, treeName will not be set. Regardeless, I recommend using isEmptyString()

Reply:
I believe the condition as written is equivalent to !isEmptyString(altElementName), but agree that the function is clearer. Will also change three similar occurrences (1 new, 2 existing) in handleFinalRequest.